blocks.md

```diff
@@ -0,0 +1,111 @@
+
+<h4>3. Plugin and block names should reflect the block's purpose.</h4>
+
+Plugin titles and block titles should describe what the block does in a way that helps users easily understand its purpose. In most cases the plugin title and the block title should be identical or very similar. Some examples of good plugin and block names include:
+
+`Rainbow Block`
+`Sepia Image Grid`
+`Business Hours Block`
+
+Please avoid plugin and block titles unrelated to the block itself, or that cannot be easily distinguished from core blocks. Some examples of unhelpful plugin and block names are things such as:
+
+`PluginCo Inc Block`
+`Widget`
+`Image Block`
+
+You must refrain from names and titles that include trademarks or names that you do not legally represent; for example you cannot name a block "Facerange Block" unless you work for Facerange and have their permission.
+
+<h4>3a. Block names should be unique and properly namespaced.</h4>
+
+The block name (meaning the [`name` parameter to `registerBlockType()`](https://developer.wordpress.org/block-editor/developers/block-api/block-registration/#block-name) and [`name` in `block.json`](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#name)) must be unique to your block. As with titles, please avoid names containing trademarks or names you do not legally represent.
+
+The namespace prefix to the block name should reflect either the plugin author, or the plugin slug. For example:
+
+`name: "my-rainbow-block-plugin/rainbow"`, or
+`name: "john-doe/rainbow"`, or
+`name: "pluginco/rainbow"`.
+
+The namespace must not be a reserved one such as `core`.
+
+<h4>4. Block Plugins must include a <code>block.json</code> file.</h4>
+
+The Block Registration RFC outlines the format of a `block.json` file: https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md
+
+Block Plugins must include a valid `block.json` file. In addition to the requirements specified in the RFC, the `block.json` must include the following attributes:
+* name
+* title
+* At least one of: script, editorScript
+* At least one of: style, editorStyle
+
+<h4>5. Block Plugins must not require payment or a paid account to function.</h4>
+
+This includes payment for premium features. You may use the donation link feature.
+
+Block plugins that do require a paid service or include upselling and premium features are still permitted in the main WordPress Plugin Directory, subject to its guidelines.
+
+<h4>6. Block Plugins should work seamlessly.</h4>
+
+Block Plugins are intended to work seamlessly and instantly when installed from the editor. That means they should not encumber the user with additional steps or prerequisites such as installing another plugin or theme, signing up for an account, or logging in or manually connecting to an external service.
+
+Block Plugins must not require another plugin in order to function.
```

Comment:
Will we revisit this later? This means people can't include WooCommerce blocks, as that requires another plugin. (We really need a plugin dependency check like we have for themes)

Reply:
Until we have a way to specify and search based on dependencies, I think it's important to prohibit plugin dependencies. Otherwise people will wind up installing blocks they can't use.

However, I have expanded and changed some of the clauses around that. In particular I've made it clear that it should be fine to make use of an existing plugin if it happens to be installed, as long as the block is also usable without it.

For example, a "One Day Only Discount Banner Block" plugin could use hooks to interact with WeeCommerce, for example to display a coupon code - as long as that feature is optional and the discount banner block is usable on sites that don't have WeeCommerce installed.

Reply:
> Until we have a way to specify and search based on dependencies, 

Yeah, that's fair. As long as we're keeping it open for if/when we get that :sigh: I really want plugin dependencies like we have for theme.